core/src/main/java/org/elasticsearch/common/io/FastStringReader.java

```diff
@@ -49,8 +47,9 @@ public FastStringReader(String s) {
      * Check to make sure that the stream has not been closed
      */
     private void ensureOpen() throws IOException {
-        if (length == -1)
+        if (closed) {
```

Comment:
FastStringReader uses a length equal to -1 to indicate that the reader has been closed. But once closed, the CharSequence's length() method reports an invalid value making use of subSequence(int,int) method fail. I fixed that by adding a closed flag instead of relying on length value.

Reply:
It has been caught by tests, because Jackson 2.9.2 is more descriptive than 2.8.6 when reporting JSON location in exception messages and uses CharSequence's length() method [here](https://github.com/FasterXML/jackson-core/commit/ce193ae9f303a26ad40b0e65c3425b8e054c237e#diff-5c5372d0a67f94e3ecbcadc081d956c5R158)

Reply:
Makes sense to me.